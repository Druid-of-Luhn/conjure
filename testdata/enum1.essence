language Essence 2.0

letting e be enum { foo, bar, baz }

find x,y : e

such that x != y
