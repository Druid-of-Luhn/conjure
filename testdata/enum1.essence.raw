Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Letting
                         , "e"
                         , DomainEnum
                             { enums = [ "foo"
                                       , "bar"
                                       , "baz"
                                       ]
                             , representation = Nothing
                             }
                         )
                       , ( Find
                         , "x"
                         , Identifier "e"
                         )
                       , ( Find
                         , "y"
                         , Identifier "e"
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Neq
                      [ Identifier "x"
                      , Identifier "y"
                      ]
                  ]
  , metadata = []
  }