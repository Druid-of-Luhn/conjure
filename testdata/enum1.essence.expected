language Essence 2.0

letting e be domain enum {foo, bar, baz}
find x : e
find y : e

such that
    x != y
